### PR TITLE
Amlogic: eliminate config prompts; align compression and drivers with other projects

### PR DESCRIPTION
**Pull request by MilhouseVH**
Build tested only (both AMLGX and AMLG12). Cherry pick, or merge, but either way the current Amlogic kernel config needs some attention as it's not "clean".

* I enabled `crazycat` and `dvb-latest` packages - no build issues observed

* Dragonboard uses `gz` initramfs compression, while Allwinner uses `lz4`. I flipped a coin for Amlogic.

Kernel size comparison - not much difference, but now consistent with other projects:

| Device | Without compression | With compression |
| ------ | ------------------: | ---------------: |
| AMLGX  |           8,928,504 |        8,806,313 |
| AMLG12 |           8,926,833 |        8,804,677 |

* I ran `make oldconfig` on the currently merged config and accepted defaults for all prompts

* Dropped redundant property, kernel unpacks fine without it

* I haven't bumped the kernel to the 5.3 release version

### CODE REVIEW

**Comment by MilhouseVH** [comment]
I've dropped the commit that removed `PKG_SOURCE_NAME` - the (entirely valid) purpose of this property is to ensure that a unique filename is used when downloading the tarball and avoid clashing with any other similarly named (but different) tarball... we could possibly make `PKG_SOURCE_NAME` a default option for the `linux` package rather than adding it for specific projects. Separate PR...

File: projects/Amlogic/linux/linux.aarch64.conf [hunk 1]
```diff
@@ -3111,32 +3109,32 @@ CONFIG_VIDEO_CX25840=m
 CONFIG_CXD2880_SPI_DRV=m
 # end of Media SPI Adapters
 
-CONFIG_MEDIA_TUNER=m
+CONFIG_MEDIA_TUNER=y
```
Comment (CvH) [inline projects/Amlogic/linux/linux.aarch64.conf]: any reason for this ?
every other project builds the dvb drivers as modules

Comment (jernejsk) [reply projects/Amlogic/linux/linux.aarch64.conf]: That's result of `make oldconfig` so there must be something which is set to `=y` and thus whole chain becomes `=y` too.

Comment (MilhouseVH) [reply projects/Amlogic/linux/linux.aarch64.conf]: Yep, I just accepted the defaults which is what will be used during the build (assuming the build doesn't halt completely waiting for an answer to a prompt, and is quite likely). If these extra options need fine tuning then I would defer that to someone more intimately involved with the Amlogic project.

BTW Allwinner (both `arm` and `aarch64` configs) specifies `CONFIG_MEDIA_TUNER=y`, while all other projects (including Amlogic) use `CONFIG_MEDIA_TUNER=m` (which is now being set to `=y` in this PR). I'm not sure what is forcing this option to `=y`, or if it should be `=m`.

Comment (MilhouseVH) [reply projects/Amlogic/linux/linux.aarch64.conf]: Further to my previous comment `=y` is the default for `CONFIG_MEDIA_TUNER`, and setting it to `=m` has no effect (it remains `=y`), so something else is setting it `=y`...

**Comment by MilhouseVH** [comment]
Setting the following option to `=m` allows `CONFIG_MEDIA_TUNER` (and all `CONFIG_MEDIA_TUNER_*` options) to build as modules:
```
CONFIG_MEDIA_SUPPORT=m
```

@jernejsk this option may need to be set for Allwinner too.

**Comment by jernejsk** [comment]
Nope, I set Cedrus VPU driver as built-in (`=y`) because I don't want to be overridden in any case. This driver has following dependencies: STAGING [=y] && STAGING_MEDIA [=y] && MEDIA_SUPPORT [=y] && VIDEO_SUNXI [=y] && VIDEO_DEV [=y] && VIDEO_V4L2 [=y] && MEDIA_CONTROLLER [=y] && HAS_DMA [=y] && OF [=y] && MEDIA_CONTROLLER_REQUEST_API [=y]

I guess that rules out some features like crazycat and dvb-latest, but meh.

**Comment by chewitt** [comment]
I'm not 100% sold on the need to include DVB drivers "because we use them on other projects" but I'd rather merge and prove there either are/are-not driver issues so things can be reverted with a reason than be forever subject to the config police. I'd still like some ability to focus DVB support on USB drivers that Amlogic users might use, as there's really no practical reason to include all the PCI/PCIe drivers and firmware that will never be used.

**Pull request merged**